junit-plugin/src/main/scala/org/scalajs/junit/plugin/ScalaJSJUnitPlugin.scala

```diff
@@ -0,0 +1,544 @@
+        bootSym.flags += Flags.MODULE
+        bootSym.withoutAnnotations
+        bootSym.setName(bootName)
+        val newClazzInfo = {
+          val newParentsInfo = List(
+            definitions.ObjectTpe,
+            jUnitTestMetadataType
+          )
+          val decls = bootSym.info.decls
+          decls.enter(getJUnitMetadataDef.symbol)
+          decls.enter(newInstanceDef.symbol)
+          decls.enter(invokeJUnitMethodDef.symbol)
+          decls.enter(invokeJUnitMethodOnInstanceDef.symbol)
+          ClassInfoType(newParentsInfo, decls, bootSym.info.typeSymbol)
+        }
+        bootSym.setInfo(newClazzInfo)
+        scalaJSPlugin.PrepInteropComponent.registerModuleExports(bootSym)
+        bootClazz.setSymbol(bootSym)
+
+        bootClazz
+      }
+
+      def jUnitAnnotatedMethods(sym: Symbol): List[MethodSymbol] = {
+        sym.selfType.members.collect {
+          case m: MethodSymbol if hasJUnitMethodAnnotation(m) => m
+        }.toList
+      }
+
+      def getMethods(clazzDef: ClassDef): List[DefDef] =
+        clazzDef.impl.body.collect { case dDef: DefDef => dDef }
+
+      /**
+       * This method generates a method that invokes a test method in the module
+       * given its name. These methods have no parameters.
+       *
+       * Example:
+       * {{{
+       * object Foo {
+       *   @BeforeClass def bar(): Unit
+       *   @AfterClass def baz(): Unit
+       * }
+       * object Foo\$scalajs\$junit\$bootstrapper
+       *   // This is the method generated by mkInvokeJUnitMethodOnModuleDef
+       *   def invoke(methodName: String): Unit = {
+       *     if (methodName == "bar") Foo.bar()
+       *     else if (methodName == "baz") Foo.baz()
+       *     else throw new NoSuchMethodException(methodName + " not found")
+       *   }
+       * }
+       * }}}
+       */
+      def mkInvokeJUnitMethodOnModuleDef(methods: List[MethodSymbol],
+          bootSym: Symbol, mlClassSym: Option[Symbol]): DefDef = {
+
+        val invokeJUnitMethodSym = bootSym.newMethod(newTermName("invoke"))
+
+        val paramSym =
+          invokeJUnitMethodSym.newValueParameter(newTermName("methodName"))
+        paramSym.setInfo(typeOf[java.lang.String])
+
+        invokeJUnitMethodSym.setInfo(
+          MethodType(List(paramSym), definitions.UnitTpe))
+
+        val methodNotFound = {
+          val msg = gen.mkMethodCall(paramSym, nme.PLUS, Nil,
+            List(Literal(Constant(" not found"))))
+          val exception = mkNewInstance[NoSuchMethodException](List(msg))
+          Throw(exception)
+        }
+
+        val invokeJUnitMethodRhs0 = {
+          methods.foldRight[Tree](methodNotFound) {
+            case (methodSymbol, acc) =>
+              val methodSymbolLocal = {
+                mlClassSym.fold[Symbol](methodSymbol) { sym =>
+                  methodSymbol.cloneSymbol(newOwner = sym)
+                }
+              }
+              val headMethodId = Literal(Constant(methodSymbol.nameString))
```

Comment:
Doesn't `nameString` *decode* the name? If it does, you need to use `.name.toString` instead.

Add a test case for a method whose name contains a `+` or any other symbol that needs to be encoded.